src/llama.cpp

```diff
@@ -11106,6 +11123,12 @@ struct llm_build_context {
 
         // lm_head
         cur = ggml_mul_mat(ctx0, model.output, cur);
+
+        // final logit soft-capping
+        cur = ggml_scale(ctx0, cur, 1.0f / hparams.f_final_logit_softcapping);
```

Comment:
@ggerganov 
Total nitpick that probably should be ignored. I came here from curiosity, and I know this is merged by now and I have absolutely no place to comment. But 
Isn’t this similar logic to lines 7594-7596? 

I know the “rule of 3” but shouldn’t this be extracted to apply_softcap method? Or is it too trivial to need that? (Risk is someone down the road might tweak one place and not be aware they need to do so in the other… )